### PR TITLE
fix TimeSeries.pd_series not retaining original name

### LINKED ISSUE #2101: [BUG] TimeSeries.pd_series does not retain original name
**Describe the bug**

The new `Pandas.Series` instance returned by the `TimeSeries.pd_series` method does not retain the original name of the `TimeSeries` object.

**To Reproduce**
```
>>> import pandas as pd
>>> from darts.timeseries import TimeSeries
>>> idx = pd.date_range("2018-01-01", periods=5, freq="D")
>>> ts = pd.Series(range(len(idx)), index=idx, name="series_name")
>>> ts
time
2018-01-01    0
2018-01-02    1
2018-01-03    2
2018-01-04    3
2018-01-05    4
Freq: D, Name: series_name, dtype: int64
>>> series = TimeSeries.from_series(ts)
>>> series.pd_series()
time
2018-01-01    0.0
2018-01-02    1.0
2018-01-03    2.0
2018-01-04    3.0
2018-01-05    4.0
Freq: D, dtype: float64
```

**Expected behavior**

I expected the `TimeSeries.pd_series` method to return a new `Pandas.Series` instance with the name of the original `TimeSeries` object (`series`). Shouldn't this be the expected behavior?

**System (please complete the following information):**

 - Python version: 3.10
 - darts version 0.25.0

### PR DESCRIPTION
**Pull request by fmerinocasallo**
Fixes #2101.

### Summary

In this new version of the `TimeSeries.pd_series` method, the newly created `Panda.Series` instance to be returned takes as its name the one from the `TimeSeries` object (stored as the first item in the iterable attribute `self.components`).

### Other Information

This new version of the `TimeSeries.pd_series` method returns a new `Pandas.Series` instance with the name of the original `TimeSeries` object (`series`).

```
>>> import pandas as pd
>>> from darts.timeseries import TimeSeries
>>> idx = pd.date_range("2018-01-01", periods=5, freq="D")
>>> ts = pd.Series(range(len(idx)), index=idx, name="series_name")
>>> ts
2018-01-01    0
2018-01-02    1
2018-01-03    2
2018-01-04    3
2018-01-05    4
Freq: D, Name: series_name, dtype: int64
>>> series = TimeSeries.from_series(ts)
>>> series.pd_series()
time
2018-01-01    0.0
2018-01-02    1.0
2018-01-03    2.0
2018-01-04    3.0
2018-01-05    4.0
Freq: D, Name: series_name, dtype: float64
```

### CODE REVIEW

File: darts/timeseries.py [hunk 1]
```diff
@@ -1472,10 +1472,10 @@ def pd_series(self, copy=True) -> pd.Series:
         self._assert_deterministic()
         if copy:
             return pd.Series(
-                self._xa[:, 0, 0].values.copy(), index=self._time_index.copy()
+                self._xa[:, 0, 0].values.copy(), index=self._time_index.copy(), name=self.components.array[0]
```
Comment (dennisbader) [inline darts/timeseries.py:1475]:
```suggestion
                self._xa[:, 0, 0].values.copy(), index=self._time_index.copy(), name=self.components[0]
```

File: darts/timeseries.py [hunk 2]
```diff
@@ -1472,10 +1472,10 @@ def pd_series(self, copy=True) -> pd.Series:
...
             )
         else:
-            return pd.Series(self._xa[:, 0, 0].values, index=self._time_index)
+            return pd.Series(self._xa[:, 0, 0].values, index=self._time_index, name=self.components.array[0])
```
Comment (dennisbader) [inline darts/timeseries.py:1478]:
```suggestion
            return pd.Series(self._xa[:, 0, 0].values, index=self._time_index, name=self.components[0])
```

**Review by dennisbader — CHANGES_REQUESTED** [review]
Hi @fmerinocasallo, thanks for this fix and the well structured issue 🚀 

Just a minor suggestion, then we can merge :)

**Comment by fmerinocasallo** [comment]
Hey @dennisbader, thanks for your suggestion! :smile: You are absolutely right. For some unknown reason, I assumed we could not directly access items in the `components` attribute from TimeSeries instances :sweat_smile: I have already commited your suggested changes :ok_hand:

**Review by dennisbader — APPROVED** [review]
Looks great now, thanks @fmerinocasallo 🚀

**Comment by dennisbader** [comment]
You can reformat the code with black to resolve the failing lint check. 

Also if you plan on contributing more in the future, you could also set up the pre-commit hooks described in our CONTRIBUTING.md that will automatically check and reformat the code

**Comment by fmerinocasallo** [comment]
I reformatted the code with black to resolve the failing link check, as instructed. However, there are a couple of tests that are still not successful. Is there something else I can do on my side to resolve this?

**Comment by fmerinocasallo** [comment]
I am afraid I could use some help to fix that unit test failing, as I don't know how to proceed :sweat_smile: I would appreciate if you continue pointing me on the right direction :pray: This way, I can learn to fix it by myself and do it autonomously next time :smile:

**Review by dennisbader — APPROVED** [review]
Hi @fmerinocasallo and thanks again for the contribution. 🚀 💯 
I pushed to fix to the failing unit test, since we want to make a patch release soon.

If you want, you can check out what I did in my latest commits to see how I normally do this (in case you want to contribute more in the future :) ).

**Comment by fmerinocasallo** [comment]
Thanks for your patience and assistance, @dennisbader :) I am sorry for my slow progress on this front :sweat:

I have checked out your latest commits and understood the issue regarding the row indices of static cov dataframes I was struggling with. I thought the series.quantile_timeseries() was supposed to add the string literal `"_quantiles"` to component names. Now I realize `"_quantiles"` was a placeholder (as in `"_{quantiles}"`) :sweat_smile: 

I will keep an eye on the project in case I can continue to contribute in the future :wink:

**Pull request merged**